Source/DafnyDriver/DafnyDoc.cs

```diff
@@ -817,6 +817,11 @@ class DafnyDoc {
         details.Append(space4).Append(Keyword("requires")).Append(" ").Append(Code(req.E.ToString())).Append(br).Append(eol);
         some = true;
       }
+      if (m.Reads.Count > 0) {
+        var list = String.Join(", ", m.Reads.Select(e => Code(e.OriginalExpression.ToString() + (e.FieldName != null ? "`" + e.FieldName : ""))));
```

Comment:
It feels like you can avoid some code duplication by creating at least a local function to render both reads and modifies clause, i.e. to extract `.Select(e => Code(e.OriginalExpression.ToString() + (e.FieldName != null ? "`" + e.FieldName : ""))));` both for the modifies and reads clause.
But as it's the doc tool, don't worry about that too much.

Reply:
Yeah there are small bits of duplication in this that weren't quite painful to bother with. But I am going to try to improve the larger bits.